tests/API/Management/UsersTestMocked.php

```diff
@@ -0,0 +1,525 @@
+        $this->assertEquals( self::$expectedTelemetry, $headers['Auth0-Client'][0] );
+    }
+
+    /**
+     * Test an update user call.
+     *
+     * @return void
+     *
+     * @throws \Exception Should not be thrown in this test.
+     */
+    public function testThatUpdateUserRequestIsFormedProperly()
+    {
+        $api = new MockManagementApi( [ new Response( 200, self::$headers ) ] );
+
+        $api->call()->users->update(
+            '__test_user_id__',
+            [
+                'given_name' => '__test_given_name__',
+                'user_metadata' => [
+                    '__test_meta_key__' => '__test_meta_value__'
+                ]
+            ]
+        );
+
+        $this->assertEquals( 'PATCH', $api->getHistoryMethod() );
+        $this->assertEquals( 'https://api.test.local/api/v2/users/__test_user_id__', $api->getHistoryUrl() );
+        $this->assertEmpty( $api->getHistoryQuery() );
+
+        $headers = $api->getHistoryHeaders();
+        $this->assertEquals( 'Bearer __api_token__', $headers['Authorization'][0] );
+        $this->assertEquals( self::$expectedTelemetry, $headers['Auth0-Client'][0] );
+
+        $body = $api->getHistoryBody();
+        $this->assertArrayHasKey( 'given_name', $body );
+        $this->assertEquals( '__test_given_name__', $body['given_name'] );
+        $this->assertArrayHasKey( 'user_metadata', $body );
+        $this->assertArrayHasKey( '__test_meta_key__', $body['user_metadata'] );
+        $this->assertEquals( '__test_meta_value__', $body['user_metadata']['__test_meta_key__'] );
+    }
+
+    public function testThatExceptionIsThrownWhenConnectionIsMissing()
+    {
+        $api = new Management( uniqid(), uniqid() );
+
+        try {
+            $api->users->create( [] );
+            $exception_message = '';
+        } catch (\Exception $e) {
+            $exception_message = $e->getMessage();
+        }
+
+        $this->assertContains( 'Missing required "connection" field', $exception_message );
+    }
+
+    public function testThatExceptionIsThrownWhenSmsConnectionHasNoPhoneNumber()
+    {
+        $api = new Management( uniqid(), uniqid() );
+
+        try {
+            $api->users->create( [ 'connection' => 'sms' ] );
+            $exception_message = '';
+        } catch (\Exception $e) {
+            $exception_message = $e->getMessage();
+        }
+
+        $this->assertContains( 'Missing required "phone_number" field for sms connection', $exception_message );
+    }
+
+    public function testThatExceptionIsThrownWhenConnectionHasNoEmailAddress()
+    {
+        $api = new Management( uniqid(), uniqid() );
+
+        try {
+            $api->users->create( [ 'connection' => 'email' ] );
+            $exception_message = '';
+        } catch (\Exception $e) {
+            $exception_message = $e->getMessage();
+        }
+
+        $this->assertContains( 'Missing required "email" field', $exception_message );
```

Comment:
are these messages triggered from your SDK or the real server?

Reply:
SDK